types.go

```diff
@@ -737,11 +786,11 @@ func writePLPType(w io.Writer, ti typeInfo, buf []byte) (err error) {
 	}
 }
 
-func readVarLen(ti *typeInfo, r *tdsBuffer, c *cryptoMetadata) {
+func readVarLen(ti *typeInfo, r *tdsBuffer, c *cryptoMetadata, guidConversion bool) {
```

Comment:
there might be other variables that control these functions in the future.
Instead of a simple `bool` maybe we should have some kind of encodeparameters type whose only member today is this boolean but where we can add more flags in the future without changing these method signatures again.